numpyro/infer/mcmc.py

```diff
@@ -635,7 +635,7 @@ def run(self, rng_key, *args, extra_fields=('diverging',), collect_warmup=False,
                 raise ValueError('`init_params` must have the same leading dimension'
                                  ' as `num_chains`.')
         assert isinstance(extra_fields, (tuple, list))
-        collect_fields = ('z',) + tuple(extra_fields) if 'z' not in extra_fields else extra_fields
+        collect_fields = set(('z', 'diverging') + tuple(extra_fields))
```

Comment:
Do we need another wrapper `tuple(...)` to convert this variable to a tuple? I am not sure if the order of `states` in `_single_chain_mcmc` method is aligned well if we use `set`.
```
collect_fn = attrgetter(*collect_fields)
...
states = dict(zip(collect_fields, states))
```

Reply:
Good point; fixed!